src/soter/soter.mk

```diff
@@ -85,6 +84,10 @@ install_soter: err $(BIN_PATH)/$(LIBSOTER_A) $(BIN_PATH)/$(LIBSOTER_SO) $(BIN_PA
 	@$(INSTALL_DATA) $(BIN_PATH)/libsoter.pc            $(DESTDIR)/$(pkgconfigdir)
 	@$(INSTALL_DATA) $(BIN_PATH)/$(LIBSOTER_A)          $(DESTDIR)/$(libdir)
 	@$(INSTALL_PROGRAM) $(BIN_PATH)/$(LIBSOTER_SO)      $(DESTDIR)/$(libdir)
+ifdef IS_MACOS
+	@install_name_tool -id "$(libdir)/$(LIBSOTER_SO)" "$(DESTDIR)/$(libdir)/$(LIBSOTER_SO)"
```

Comment:
so we actually need to use `install_name_tool`? :)

Reply:
That's a long story... To sum it up: _think different™_

If we're not going to use the dynamic libraries from the build directory then we don't need `install_name_tool`. We can use `-dylinker_install_name` linker flags to get the install names right when we build the libraries. You can still link against those libraries, but applications won't start until the libraries are actually installed into the system at the specified paths.

We did not need to use dynamic libraries during the build because all our tests are using static linkage now. However, if we are going to use dynamic linkage for tests then we can't set the install name right away (or the tests won't start).

The correct approach is to apply `install_name_tool` _after_ installation. This way the libraries keep install names like `build/libthemis.0.dylib`, and these will be recorded into test binaries. We can then start the binaries from the directory we built them in (but not the others, unfortunately). Note that this means that `libthemis.0.dylib` initially records incorrect path to `libsoter.0.dylib` which we fix during installation.

Before this PR we changed install names during the build. This also made dynamic libraries unusable for tests unless installed. As an added bonus, this means that we can correctly install the libraries into multiple locations on macOS without rebuilding them (not that anyone needed that, but that's nice).